protobuf/proto/agent.proto

```diff
@@ -80,64 +81,21 @@ message DeployRequest {
     optional string repositoryPrefix = 4; // registry repo prefix
   }
   /* InstanceConfig is set for multiple containers */
-  InstanceConfig instanceConfig = 2;
-
-  message ContainerConfig {
-    /* Container name - must have, used by everthing */
-    string name = 1;
-    /* Container prefix */
-    optional string prefix = 2;
-    message Port {
-      int32 internal = 1; // internal that is bound by the container
-      int32 external = 3; // external is docker only
-    }
-    /* container ports */
-    repeated Port ports = 3;
-    /* volume mounts in a piped format */
-    repeated string mounts = 4;
-    /* environment variables in a piped format  */
-    repeated string environments = 5;
-    /* could be enum, i'm not sure if it is in use */
-    optional string networkMode = 6;
-    enum RuntimeConfigType {
-      /* appconfig will be parsed into environment variables */
-      DOTNET_APPCONFIG = 0;
-    }
-    // runtime config type if given, magic can happen
-    optional RuntimeConfigType runtimeConfigType = 7;
-    message Expose {
-      bool public = 1; // if expose is needed
-      bool tls = 2;    // if tls is needed
-    }
-    /* exposure configuration*/
-    optional Expose expose = 8;
-
-    message ConfigContainer {
-      string image = 1;
-      string volume = 2;
-      string path = 3;
-      bool keepFiles = 4;
-    }
-    /* Config container is started before the container and contents are copied
-     * to the volume set */
-    optional ConfigContainer configContainer = 9;
-    /* userId that is used to run the container, number */
-    int64 user = 10;
-  }
+  InstanceConfig instanceConfig = 3;
 
   /* Runtime info and requirements of a container */
-  ContainerConfig containerConfig = 3;
-  optional string runtimeConfig = 4;
-  optional string registry = 5;
-  string imageName = 6;
-  string tag = 7;
+  common.ExplicitContainerConfig containerConfig = 4;
```

Comment:
I loved the `ContainerConfig` instead of `ExplicitContainerConfig`.

Reply:
If you look down a bit , there is a ContainerConfig message already

Reply:
I do not find it.

Reply:
Sorry. It is in the common.proto

Reply:
Hmmm okay, then it would be more consistent if we rename the variable to explicitContainerConfig, IMHO. But I can live with that name too. Resolved the comment.